pallets/msa/Cargo.toml

```diff
@@ -26,6 +26,7 @@ sp-std = {default-features = false, git = "https://github.com/paritytech/substra
 
 # Frequency related dependencies
 common-primitives = {default-features = false, path = "../../common/primitives"}
+common-runtime = {path = '../../runtime/common', default-features = false}
```

Comment:
Apart from all the good work

I just generally don't like a dependency on runtime code but can't help it 


I wish we had  the capability to modify checked extrinsic but without that this is good

Reply:
Agree but as you know, we tried several approaches over a long period time with different devs.  If in the future we find a better solution (this is implementation is actually pretty good and works), we can always update it.